src/components/ProfileModal/EditProfileModal.tsx

```diff
@@ -0,0 +1,230 @@
+import { Button } from "@/components/ui/button";
+import { DialogFooter } from "@/components/ui/dialog";
+import { Input } from "@/components/ui/input";
+import { Label } from "@/components/ui/label";
+import {
+  Select,
+  SelectContent,
+  SelectGroup,
+  SelectItem,
+  SelectTrigger,
+  SelectValue,
+} from "@/components/ui/select";
+
+import { userSchema } from "@/utils/types";
+import cn from "classnames";
+import { useState, useEffect } from "react";
+
+type EditProps = {
+  setProfileState: React.Dispatch<
+    React.SetStateAction<"view" | "edit" | "changePw">
+  >;
+  userData:
+    | {
+        email: string;
+        hashedPassword: string;
+        firstName: string;
+        lastName: string;
+        label: "educator" | "student" | "parent" | "administrator";
+        _id: string;
+      }
+    | undefined;
+  setUserData: React.Dispatch<
+    React.SetStateAction<
+      | {
+          email: string;
+          hashedPassword: string;
+          firstName: string;
+          lastName: string;
+          label: "educator" | "student" | "parent" | "administrator";
+          _id: string;
+        }
+      | undefined
+    >
+  >;
+  setOpen: React.Dispatch<React.SetStateAction<boolean>>;
+  editUser: (
+    data:
+      | {
+          email: string;
+          password: string;
+          oldpassword: string;
+          passwordConfirm: string;
+        }
+      | {
+          email: string;
+          hashedPassword: string;
+          firstName: string;
+          lastName: string;
+          label: "educator" | "student" | "parent" | "administrator";
+          _id: string;
+        },
+    type: "info" | "password"
+  ) => Promise<any>;
```

Comment:
This type should be inferred from the actual function definition itself. See above